### PR TITLE
Add gates

### LINKED ISSUE #387: Gates
As an elf I want there to be gates I can open, so that I have another way to help the vrplayer.

This task has the following subtasks:
- #388 Gates - Implement gates in game
- #389 Gates - Add gates to web interface

### LINKED ISSUE #388: Gates - Implement gates in game
Adds a gate entity and its functions

This is a subtask of #387 - Gates

### LINKED ISSUE #389: Gates - Add gates to web interface
The webinterface should be able to show gates

This is a subtask of #387 - Gates

### PR DESCRIPTION
**Pull request by WytzeElhorst**
Adds gate object to the game and the ability to open them on the webinterface, as well as a testclass for the gate

this closes #387, closes #388 and closes #389

### CODE REVIEW

File: src/main/java/nl/tudelft/contextproject/model/entities/Gate.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+import nl.tudelft.contextproject.Main;
+import nl.tudelft.contextproject.model.PhysicsObject;
+
+/**
+ * Class representing a gate.
+ */
+public class Gate extends Entity implements PhysicsObject {
+	private Spatial sp;
+	private RigidBodyControl rb;
+	private Boolean open;
```
Comment (ChielBruin) [inline src/main/java/nl/tudelft/contextproject/model/entities/Gate.java:20]: Add a newline here

File: src/main/java/nl/tudelft/contextproject/model/entities/Gate.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+		open = false;
+	}
+
+	@Override
+	public Spatial getSpatial() {
+		return sp;
+	}
+
+	@Override
+	public void update(float tdf) {
```
Comment (ChielBruin) [inline src/main/java/nl/tudelft/contextproject/model/entities/Gate.java]: this should be tpf

**Comment by ChielBruin** [comment]
Apart from two minor style complaints and a missing test, all code looks good and can be merged.

File: src/main/java/nl/tudelft/contextproject/model/entities/Gate.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+		if (open) {
+			if (sp.getLocalTranslation().y < 6) {
+				this.move(0, 2 * tdf, 0);
+			} else {
+				open = false;
+			}
+		} else {
+			if (sp.getLocalTranslation().y > 1) {
```
Comment (ChielBruin) [inline src/main/java/nl/tudelft/contextproject/model/entities/Gate.java]: shouldn't this be .5f?

File: src/main/java/nl/tudelft/contextproject/model/entities/Gate.java [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+	private Spatial sp;
+	private RigidBodyControl rb;
+	private Boolean open;
+	/**
+	 * Constructor for a gate.
+	 */
+	public Gate() {
+		sp = Main.getInstance().getAssetManager().loadModel("Models/door.blend");
+		sp.scale(1.2f, 2.2f, 1.2f);
+		sp.rotate(0, (float) (Math.PI), 0);
```
Comment (ChielBruin) [inline src/main/java/nl/tudelft/contextproject/model/entities/Gate.java:28]: add

``` java
sp.move(0, .5f, 0);
```

to ensure the door is loaded at the correct location when loaded from a file

File: src/main/java/nl/tudelft/contextproject/model/entities/EntityType.java [hunk 5]
```diff
@@ -29,8 +29,8 @@
 	WALLFRAME("WallFrame", 			0,  WallFrame::loadEntity),
 	TORCH("Torch", 					0,  Torch::loadEntity),
 	INVISIBLE_WALL("InvisibleWall",	11,	InvisibleWall::loadEntity),
-	DAMAGED_WALL("DamagedWall",		12,	DamagedWall::loadEntity);
-	
+	DAMAGED_WALL("DamagedWall",		12,	DamagedWall::loadEntity),
+	GATE("Torch",					14,	Torch::loadEntity);
```
Comment (ChielBruin) [inline src/main/java/nl/tudelft/contextproject/model/entities/EntityType.java]: Gate should not be a torch

**Comment by ChielBruin** [comment]
All looks good now!

File: src/main/java/nl/tudelft/contextproject/model/entities/Gate.java (hunk 1 above)
Comment (Taeir) [inline src/main/java/nl/tudelft/contextproject/model/entities/Gate.java:20]: _resolved_

File: src/main/java/nl/tudelft/contextproject/model/entities/Gate.java (hunk 4 above)
Comment (Taeir) [inline src/main/java/nl/tudelft/contextproject/model/entities/Gate.java:28]: _resolved_

**Comment by Taeir** [comment]
Looks good, nice work :).

I'm merging this.

**Pull request merged**